cpp-package/example/inference/inception_inference.cpp

```diff
@@ -0,0 +1,414 @@
+    std::cout << "inception_inference --symbol <model symbol file in json format>  "
+              << "--params <model params file> "
+              << "--image <path to the image used for prediction "
+              << "--synset file containing labels for prediction "
+              << "[--input_shape <dimensions of input image e.g \"3 224 224\"]"
+              << "[--mean file containing mean image for normalizing the input image "
+              << "[--gpu]  Specify this option if workflow needs to be run in gpu context "
+              << std::endl;
+}
+
+int main(int argc, char** argv) {
+  string model_file_json;
+  string model_file_params;
+  string synset_file = "";
+  string mean_image = "";
+  string input_image = "";
+  bool gpu_context_type = false;
+
+  std::string input_shape = "3 224 224";
+    int index = 1;
+    while (index < argc) {
+        if (strcmp("--symbol", argv[index]) == 0) {
+            index++;
+            model_file_json = argv[index];
+        } else if (strcmp("--params", argv[index]) == 0) {
+            index++;
+            model_file_params = argv[index];
+        } else if (strcmp("--synset", argv[index]) == 0) {
+            index++;
+            synset_file = argv[index];
+        } else if (strcmp("--mean", argv[index]) == 0) {
+            index++;
+            mean_image = argv[index];
+        } else if (strcmp("--image", argv[index]) == 0) {
+            index++;
+            input_image = argv[index];
+        } else if (strcmp("--input_shape", argv[index]) == 0) {
+            index++;
+            input_shape = argv[index];
+        } else if (strcmp("--gpu", argv[index]) == 0) {
+            gpu_context_type = true;
+        } else if (strcmp("--help", argv[index]) == 0) {
+            printUsage();
+            return 0;
+        }
+        index++;
+    }
+
+  if (model_file_json.empty() || model_file_params.empty() || synset_file.empty()) {
+    LG << "ERROR: Model details such as symbol, param and/or synset files are not specified";
+    printUsage();
+    return 1;
+  }
+
+  if (input_image.empty()) {
+    LG << "ERROR: Path to the input image is not specified.";
+    printUsage();
+    return 1;
+  }
+
+  std::vector<index_t> input_dimensions = getShapeDimensions(input_shape);
+
+  /*
+   * Since we are running inference for 1 image, add 1 to the input_dimensions so that
+   * the shape of input data for the model will be
+   * {no. of images, channels, height, width}
+   */
+  input_dimensions.insert(input_dimensions.begin(), 1);
+
+  Shape input_data_shape(input_dimensions);
+
+  try {
+    // Initialize the predictor object
+    Predictor predict(model_file_json, model_file_params, input_data_shape, gpu_context_type,
+                      synset_file, mean_image);
+
+    // Run the forward pass to predict the image.
+    predict.PredictImage(input_image);
+  } catch (std::runtime_error &error) {
+    LG << "Execution failed with ERROR: " << error.what();
```

Comment:
Maybe 2 exceptional cases can be combined together? If exception is `std::runtime_error` it's not an MXNet error?

Reply:
I tested by giving wrong inputs such as incorrect input dimensions or corrupted json file.  It's not obvious which type of exception is thrown by MXNet engine. It was a generic exception.